src/components/vanilla/charts/KPIChart/index.tsx

```diff
@@ -12,82 +12,92 @@ type Props = {
   title?: string;
   prefix?: string;
   suffix?: string;
-  metric: { name: string; meta?: object };
+  metric?: Measure;
+  displayMetric?: boolean;
+  dimension?: Dimension;
   dps?: number;
   fontSize?: number;
 };
 
 export default (props: Props) => {
-  
+  const { results, prevResults, prevTimeFilter, metric, displayMetric, dimension, dps, prefix, suffix } = props;
+
   const { n, percentage } = useMemo(() => {
-    if (!props.results?.data?.length || !props.metric?.name) return { percentage: null };
-    
-    const n = parseFloat(props.results.data[0][props.metric?.name] || 0);
-    const prev = parseFloat(props.prevResults?.data?.[0]?.[props.metric?.name] || 0);
+    if (dimension || !metric?.name || !props.results?.data?.length) {
+      return { percentage: null, n: null }; // Skip calculations
+    }
+
+    const n = parseFloat(results?.data?.[0]?.[metric.name] || 0);
+    const prev = parseFloat(prevResults?.data?.[0]?.[metric.name] || 0);
 
     return {
       percentage: prev ? Math.round((n / prev) * 100) - 100 : null,
       n: formatValue(n, {
         type: 'number',
-        meta: props.metric?.meta,
-        dps: props.dps
+        meta: metric?.meta,
+        dps: dps
       })
     };
-  }, [props]);
+  }, [results, prevResults, metric, dps, dimension]);
```

Comment:
<img width="1140" alt="image" src="https://github.com/user-attachments/assets/f05ae907-ecba-43a9-a81b-339936239beb">

Reply:
can we have this dependency?

Reply:
@mad-raccoon do you add props.results.data.length as a depedency?

Reply:
yes !

Reply:
[results, prevResults, metric, dps, dimension, props.results.data.length]